### PR TITLE
Fix exception logged twice in CIO launch channel

### PR DESCRIPTION
**Pull request by e5l**

### CODE REVIEW

**Comment by e5l** [comment]
Sorry, I tested and found the critical here. 
This avoids logging, but doesn't cancel launcher job - decomposition stopped working.

I'm looking for the fix.

File: ktor-io/common/src/io/ktor/utils/io/Coroutines.kt [hunk 1]
```diff
@@ -43,32 +43,6 @@ fun CoroutineScope.reader(
...
-    coroutineContext: CoroutineContext,
-    autoFlush: Boolean = false, parent: Job? = null,
-    block: suspend ReaderScope.() -> Unit
-): ReaderJob {
-    val channel = ByteChannel(autoFlush)
-    return reader(coroutineContext, channel, parent, block).also {
-        channel.attachJob(it)
-    }
-}
-
```
Comment (Ololoshechkin) [inline ktor-io/common/src/io/ktor/utils/io/Coroutines.kt:71]: Why did we need this Reader/Writer jobs before and why they are not used anymore?

Comment (e5l) [reply ktor-io/common/src/io/ktor/utils/io/Coroutines.kt:71]: That's deprecated API without `CoroutineScope` that we don't want to support

File: ktor-io/common/src/io/ktor/utils/io/Coroutines.kt [hunk 2]
```diff
@@ -115,15 +64,23 @@ private fun <S : CoroutineScope> CoroutineScope.launchChannel(
     attachJob: Boolean,
     block: suspend S.() -> Unit
 ): ChannelJob {
+    val originJob = coroutineContext[Job]
     val job = launch(context) {
         if (attachJob) {
             channel.attachJob(coroutineContext[Job]!!)
```
Comment (Ololoshechkin) [inline ktor-io/common/src/io/ktor/utils/io/Coroutines.kt:70]: is it `channel.attachJob(originJob!!)` ?

Comment (e5l) [reply ktor-io/common/src/io/ktor/utils/io/Coroutines.kt:70]: Thanks. Will fix

**Pull request merged**